diff --git a/var/spack/repos/builtin/packages/podio/package.py b/var/spack/repos/builtin/packages/podio/package.py
index 37b461b1d..ebb1c0e88 100644
--- a/var/spack/repos/builtin/packages/podio/package.py
+++ b/var/spack/repos/builtin/packages/podio/package.py
@@ -48,7 +48,7 @@ class Podio(CMakePackage):
     def cmake_args(self):
         args = [
             self.define('BUILD_TESTING', self.run_tests),
-            self.define_from_variants('ENABLE_SIO', 'sio')
+            self.define_from_variant('ENABLE_SIO', 'sio')
         ]
         return args
 
