diff --git a/var/spack/repos/builtin/packages/py-awkward1/package.py b/var/spack/repos/builtin/packages/py-awkward1/package.py
index 9e6a3fec7e..93232e920f 100644
--- a/var/spack/repos/builtin/packages/py-awkward1/package.py
+++ b/var/spack/repos/builtin/packages/py-awkward1/package.py
@@ -26,4 +26,5 @@ class PyAwkward1(PythonPackage):
     depends_on('py-numpy@1.13.1:')
     depends_on('py-pybind11')
     depends_on('rapidjson')
+    depends_on('dlpack')
     depends_on('cmake', type='build')
