diff --git a/var/spack/repos/builtin/packages/libzmq/package.py b/var/spack/repos/builtin/packages/libzmq/package.py
index dac309b699..07f5ef4ac6 100644
--- a/var/spack/repos/builtin/packages/libzmq/package.py
+++ b/var/spack/repos/builtin/packages/libzmq/package.py
@@ -64,7 +64,6 @@ class Libzmq(AutotoolsPackage):
     depends_on("libunwind", when="+libunwind")
 
     conflicts("%gcc@8:", when="@:4.2.2")
-    conflicts("%gcc@11:", when="@:4.3.2")
 
     # Fix aggressive compiler warning false positive
     patch(
@@ -80,6 +79,9 @@ class Libzmq(AutotoolsPackage):
         when="@4.3.4 %gcc@12:",
     )
 
+    # Gcc12 patch fix
+    patch('https://github.com/zeromq/libzmq/commit/a01d259db372bff5e049aa966da4efce7259af67.patch?full_index=1', sha256='edca864cba914481a5c97d2e975ba64ca1d2fbfc0044e9a78c48f1f7b2bedb6f', when='@4.3.2:4.3.4 %gcc@12:')
+
     def url_for_version(self, version):
         if version <= Version("4.1.4"):
             url = "http://download.zeromq.org/zeromq-{0}.tar.gz"
diff --git a/var/spack/repos/builtin/packages/openblas/package.py b/var/spack/repos/builtin/packages/openblas/package.py
index 099d00fb..0b247175 100644
--- a/var/spack/repos/builtin/packages/openblas/package.py
+++ b/var/spack/repos/builtin/packages/openblas/package.py
@@ -445,12 +445,7 @@ def libs(self):
 
     @property
     def build_targets(self):
-        targets = ["libs", "netlib"]
-
-        # Build shared if variant is set.
-        if "+shared" in self.spec:
-            targets += ["shared"]
-
+        targets = ["all"]
         return self.make_defs + targets
 
     @run_after("build")
diff --git a/var/spack/repos/builtin/packages/rivet/package.py b/var/spack/repos/builtin/packages/rivet/package.py
index 6ea0deb3..7d5c535c 100644
--- a/var/spack/repos/builtin/packages/rivet/package.py
+++ b/var/spack/repos/builtin/packages/rivet/package.py
@@ -18,7 +18,6 @@ class Rivet(AutotoolsPackage):
 
     tags = ["hep"]
 
-    version("3.1.7b", tag="rivet-3.1.7b")
     version("3.1.7", sha256="27c7dbbcb5fd7ee81caf136daf4e960bca0ec255d9fa1abe602f4d430861b27a")
     version("3.1.6", sha256="1cf6ebb6a79d181c441d1d0c7c6d623c423817c61093f36f21adaae23e679090")
     version("3.1.4", sha256="37edc80a2968ce1031589e43ba6b492877ca7901bea38f8bb7536a5c8cf8100d")
