diff --git a/packages/k4simgeant4/package.py b/packages/k4simgeant4/package.py
index 4f7babe..135cf0a 100644
--- a/packages/k4simgeant4/package.py
+++ b/packages/k4simgeant4/package.py
@@ -24,8 +24,6 @@ class K4simgeant4(CMakePackage, Key4hepPackage):
     depends_on('k4fwcore@1.0:')
     depends_on('geant4')
     depends_on('edm4hep')
-
-
     depends_on("g4ensdfstate")
 
 
@@ -39,6 +37,16 @@ class K4simgeant4(CMakePackage, Key4hepPackage):
         spack_env.prepend_path('PYTHONPATH', self.prefix.python)
         spack_env.prepend_path("PATH", self.prefix.scripts)
         spack_env.set("K4SIMGEANT4", self.prefix.share.k4SimGeant4)
+        install_path = join_path(self.spec['g4ensdfstate'].prefix.share,
+                                 'data', 'G4ENSDFSTATE{0}'
+                                 .format(self.spec['g4ensdfstate'].version))
+        spack_env.set('G4ENSDFSTATEDATA', install_path)
+
+    def setup_build_environment(self, spack_env):
+        install_path = join_path(self.spec['g4ensdfstate'].prefix.share,
+                                 'data', 'G4ENSDFSTATE{0}'
+                                 .format(self.spec['g4ensdfstate'].version))
+        spack_env.set('G4ENSDFSTATEDATA', install_path)
 
     def setup_dependent_build_environment(self, spack_env, dependent_spec):
         spack_env.prepend_path('PYTHONPATH', self.prefix.python)
diff --git a/packages/k4reccalorimeter/package.py b/packages/k4reccalorimeter/package.py
index 55f474e..ecc5837 100644
--- a/packages/k4reccalorimeter/package.py
+++ b/packages/k4reccalorimeter/package.py
@@ -10,7 +10,7 @@ class K4reccalorimeter(CMakePackage, Key4hepPackage):
 
     maintainers = ['vvolkl']
 
-    version('master', branch='master')
+    version('main', branch='main')
     version("0.1.0pre05", tag="v0.1.0pre05")
 
     generator = 'Ninja'
@@ -25,7 +25,7 @@ class K4reccalorimeter(CMakePackage, Key4hepPackage):
             multi=False,
             description='Use the specified C++ standard when building.')
 
-    depends_on('ninja', type='build')
+    depends_on('ninja')
     depends_on("edm4hep")
     depends_on('k4fwcore@1:')
     depends_on("dd4hep +geant4")
diff --git a/packages/kkmcee/package.py b/packages/kkmcee/package.py
index a673c5c..4f02816 100644
--- a/packages/kkmcee/package.py
+++ b/packages/kkmcee/package.py
@@ -26,7 +26,6 @@ class Kkmcee(AutotoolsPackage):
     depends_on('m4',       type='build')
     depends_on('root')
 
-    patch('gcc4.patch')
     patch('gcc6.patch')
     patch('gcc5.patch')
     patch('KKMCee-dev-4.30.patch', level=0, when='@:4.30')
diff --git a/packages/larcontent/package.py b/packages/larcontent/package.py
index 8a04639..30c5502 100644
--- a/packages/larcontent/package.py
+++ b/packages/larcontent/package.py
@@ -41,6 +41,8 @@ class Larcontent(CMakePackage):
         args = [
                 '-DCMAKE_MODULE_PATH=%s' % self.spec["pandorapfa"].prefix.cmakemodules,
                 "-DCMAKE_CXX_FLAGS=-std=c++17"]
+        if self.spec.satisfies('%clang'):
+            args.append('-DCMAKE_CXX_FLAGS=-Wno-error')
         return args
 
     def url_for_version(self, version):
diff --git a/packages/lccontent/package.py b/packages/lccontent/package.py
index 0161946..f7f47cd 100644
--- a/packages/lccontent/package.py
+++ b/packages/lccontent/package.py
@@ -36,6 +36,8 @@ class Lccontent(CMakePackage):
                 '-DCMAKE_CXX_STANDARD=17',
                 '-DCMAKE_MODULE_PATH=%s' % self.spec["pandorapfa"].prefix.cmakemodules
         ]
+        if self.spec.satisfies('%clang'):
+            args.append('-DCMAKE_CXX_FLAGS=-Wno-error')
         return args
 
     def url_for_version(self, version):
