diff --git a/packages/kaltest/package.py b/packages/kaltest/package.py
index 283d8db..f7d3258 100644
--- a/packages/kaltest/package.py
+++ b/packages/kaltest/package.py
@@ -5,7 +5,8 @@
 
 
 from spack import *
-from spack.pkg.k4.Ilcsoftpackage import Ilcsoftpackage, k4_add_latest_commit_as_version
+from spack.pkg.k4.Ilcsoftpackage import Ilcsoftpackage, \
+    k4_add_latest_commit_as_version, ilc_url_for_version
 
 
 class Kaltest(CMakePackage, Ilcsoftpackage):
