--- package.py	2020-09-12 21:28:00.718166001 +0800
+++ package.py.new	2020-09-12 22:08:33.473420301 +0800
@@ -29,8 +29,11 @@
 
     def cmake_args(self):
         args = [
-                '-DCMAKE_MODULE_PATH=%s' % self.spec["pandorapfa"].prefix.cmakemodules,
-                "-DCMAKE_CXX_FLAGS=-std=c++17"]
+                '-DCMAKE_CXX_STANDARD=17',
+                '-DCMAKE_MODULE_PATH=%s' % self.spec["pandorapfa"].prefix.cmakemodules
+        ]
+        if self.spec.satisfies('%gcc@10:'):
+            args.append('-DCMAKE_CXX_FLAGS=-Wno-int-in-bool-context')
         return args
 
     def url_for_version(self, version):
